pandas/core/series.py

```diff
@@ -1446,6 +1446,50 @@ def to_string(self, buf=None, na_rep='NaN', float_format=None, header=True,
     def iteritems(self):
         """
         Lazily iterate over (index, value) tuples.
+
+        This method returns a zip of tuples (index, value). This is useful When
+        one want to create new series from the values of an old one. Be aware
+        that this might not the fastest way of creating new series.
+
+        Returns
+        -------
+        zip
```

Comment:
I wonder if people have thoughts here. I guess `zip` is technically a type... But I think Iterable is probably better-known.